support/ci/rustfmt.sh

```diff
@@ -2,31 +2,16 @@
 
 set -euo pipefail
 
-source ./support/ci/shared.sh
+d=$(dirname "${BASH_SOURCE[0]}")
+source "$d/shared.sh"
 
-# Due to the nature of nightly rust, sometimes changes will break rustfmt's
-# usage of rustc. If this happens, nightly rust won't include rustfmt,
-# and we need to automatically fall back to a version that does include it.
-# Note that we begin with 1 day ago, since nightly packages can sometimes not
-# exist when this script runs if we leave it at 0.
-max_days=90
+# Thanks to the magic of global variables, this value will get overwritten when
+# install_rustfmt runs. But, if we don't declare it up front here, shellcheck
```

Comment:
`install_rustfmt` could use a local with a different name?

Reply:
Well, we _want_ it to get overwritten though, because it's used a few lines down for the `cargo_fmt` variable.  What I _really_ want is proper return values in functions, which would eliminate the need for this. Without this line, shellcheck errored because `cargo_fmt="cargo +$toolchain fmt --all -- --check"` referenced `$toolchain`, which shellcheck couldn't see (because that variable was defined for the first time inside the `install_rustfmt` function.

Reply:
Ha ha, yeah. Return values aren't a thing.